README.md

```diff
@@ -1,3 +1,132 @@
+  #
+  # We also prevent pull_request_target, since that runs arbitrary code:
+  #   https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+  attribute_condition = "attribute.event_name != \"pull_request_target\" && attribute.repository_owner_id == \"${var.github_owner_id}\" && attribute.repository_id == \"${var.github_repository_id}\""
+
+  oidc {
+    issuer_uri = "https://token.actions.githubusercontent.com"
+  }
+
+  depends_on = [
+    google_iam_workload_identity_pool.github_pool
+  ]
+}
+
+# Service account
+resource "google_service_account" "wif_service_account" {
+  project = var.project_id
+
+  account_id   = "${substr(var.name, 0, 19)}-${random_id.default.hex}-wif-sa" # 30 character limit
+  display_name = "${var.name} WIF Service Account"
+}
+
+resource "google_service_account_iam_member" "wif_github_iam" {
+  service_account_id = google_service_account.wif_service_account.name
+  role               = "roles/iam.workloadIdentityUser"
+  member             = "principalSet://iam.googleapis.com/${google_iam_workload_identity_pool.github_pool.name}/*"
+}
+
+# Grant permissions required for operation AOD. This is just an example, your team is responsible for granting the right permissions.
+resource "google_project_iam_member" "wif_service_account_iam" {
+  for_each = toset([
+    "roles/resourcemanager.projectIamAdmin"
+  ])
+
+  project = var.project_id
+
+  role   = each.key
+  member = google_service_account.wif_service_account.member
+}
+
+# Output below values for step 3.
+output "wif_provider_name" {
+  description = "The Workload Identity Federation provider name."
+  value       = google_iam_workload_identity_pool_provider.github_provider.name
+}
+
+output "service_account_email" {
+  description = "WIF service account identity email address."
+  value       = google_service_account.wif_service_account.email
+}
+```
+
+3.  Follow steps [here](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository) to set the repository variables for WORKLOAD_IDENTITY_PROVIDER and SERVICE_ACCOUNT with outputs from step 2.
+
+4. (Optional) Create labels to be used as IAM permission expiration. IAM permissions granted via AOD will have a default 2 hour expiration when the request is approved. To use custom expirations, follow steps [here](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#creating-a-label) to create duration labels. "duration-4h" is an example of valid duration label.
+
+5. It is critical to enable the following repo settings:
+  - Disable forking
+  - Branch protection on main branch
+  - Require approvals
+  - Require review from Code Owners
+  - Disallow specific actors to bypass required pull requests
+  - Require status checks to pass before merging
+  - Require signed commits
+  - Disallow force pushes
+  - Disallow deletions
+  - Set up CODEOWNERS with the group to approve AOD requests
```

Comment:
Add a section below about adding code other than AOD. Could be a follow up to fill in the content